src/core_plugins/timelion/public/vis/timelion_request_handler.js

```diff
@@ -41,7 +41,7 @@ const TimelionRequestHandlerProvider = function (Private, Notifier, $http) {
           sheet: [expression],
           extended: {
             es: {
-              filter: buildEsQuery(aggs.indexPattern, [query], filters)
+              filter: buildEsQuery(undefined, [query], filters)
```

Comment:
the index pattern is actually defined in the expression (and there could be multiple of them) ...
so the aggs.indexPattern was not set anyway. In the future we would need to execute this on the server side with appropriate indexPatterns in order to support KQL fully